backend/driver/driver.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package driver
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	lmevalservicev1beta1 "github.com/foundation-model-stack/fms-lm-eval-service/api/v1beta1"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	client "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(lmevalservicev1beta1.AddToScheme(scheme))
+}
+
+type DriverOption struct {
+	Context      context.Context
+	JobNamespace string
+	JobName      string
+	ConfigMap    string
+	OutputPath   string
+	Logger       logr.Logger
+	Args         []string
+}
+
+type Driver interface {
+	Run() error
+}
+
+type driverImpl struct {
+	client client.Client
+	job    lmevalservicev1beta1.EvalJob
+	Option *DriverOption
+}
+
+func NewDriver(opt *DriverOption) (Driver, error) {
+	if opt == nil {
+		return nil, nil
+	}
+
+	if opt.Context == nil {
+		return nil, fmt.Errorf("context is nil")
+	}
+
+	if opt.JobNamespace == "" || opt.JobName == "" {
+		return nil, fmt.Errorf("JobNamespace or JobName is empty")
+	}
+
+	return &driverImpl{Option: opt}, nil
+}
+
+// Run implements Driver.
+func (d *driverImpl) Run() error {
+	config, err := ctrl.GetConfig()
+	if err != nil {
+		return err
+	}
+
+	client, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+
+	d.client = client
+	if err := d.getJob(); err != nil {
+		return err
+	}
+
+	if err := d.updateStatus(lmevalservicev1beta1.RunningJobState); err != nil {
```

Comment:
I think this is going to pose a problem for running jobs without RBACs. There will definitely be users that want to do this, so I think we may need to consider if we can publish status updates to some medium that would be visible to the `controller` for polling. This might just be logs?

Reply:
for now, if we want to avoid RBACs, the solutions I have on top of my mind for now are:
- a polling mechanism on the controller which I don't like
- internal API calls for the driver to update the status and upload logs

Let me also create an issue to track this
#10